pkg/api-server/config_ws_test.go

```diff
@@ -30,326 +30,8 @@ var _ = Describe("Config WS", func() {
-          "store": {
-            "kubernetes": {
-              "systemNamespace": "kuma-system"
-            },
-            "postgres": {
-              "connectionTimeout": 5,
-              "dbName": "kuma",
-              "host": "127.0.0.1",
-              "maxIdleConnections": 0,
-              "maxOpenConnections": 0,
-              "password": "*****",
-              "port": 15432,
-              "maxReconnectInterval": "1m0s",
-              "minReconnectInterval": "10s",
-              "maxIdleConnections": 50,
-              "maxOpenConnections": 50,
-              "tls": {
-                "certPath": "",
-                "keyPath": "",
-                "mode": "disable",
-                "caPath": ""
-              },
-              "user": "kuma"
-            },
-            "cache": {
-              "enabled": true,
-              "expirationTime": "1s"
-            },
-            "upsert": {
-              "conflictRetryBaseBackoff": "100ms",
-              "conflictRetryMaxTimes": 5
-            },
-            "type": "memory",
-            "unsafeDelete": false
-          },
-          "xdsServer": {
-            "dataplaneConfigurationRefreshInterval": "1s",
-            "dataplaneStatusFlushInterval": "10s",
-            "dataplaneDeregistrationDelay": "10s",
-            "nackBackoff": "5s"
-          },
-          "diagnostics": {
-            "serverPort": 5680,
-            "debugEndpoints": false
-          },
-          "access": {
-            "type": "static",
-            "static": {
-              "adminResources": {
-                "users": ["mesh-system:admin"],
-                "groups": ["mesh-system:admin"]
-              },
-              "generateDpToken": {
-                "users": ["mesh-system:admin"],
-                "groups": ["mesh-system:admin"]
-              },
-              "generateUserToken": {
-                "users": ["mesh-system:admin"],
-                "groups": ["mesh-system:admin"]
-              },
-              "generateZoneToken": {
-                "users": ["mesh-system:admin"],
-                "groups": ["mesh-system:admin"]
-              },
-              "viewConfigDump": {
-                "users": [ ],
-                "groups": ["mesh-system:unauthenticated","mesh-system:authenticated"]
-              },
-              "viewStats": {
-                "users": [ ],
-                "groups": ["mesh-system:unauthenticated","mesh-system:authenticated"]
-              },
-              "viewClusters": {
-                "users": [ ],
-                "groups": ["mesh-system:unauthenticated","mesh-system:authenticated"]
-              }
-            }
-          },
-          "experimental": {
-            "gatewayAPI": false,
-            "kubeOutboundsAsVIPs": false
-          }
-        }
-		`, cfg.HTTP.Port, cfg.HTTPS.Port)
 		// when
-		Expect(body).To(MatchJSON(json))
+		parsedCfg := kuma_cp.Config{}
+		Expect(yaml.Unmarshal(body, &parsedCfg)).To(Succeed())
```

Comment:
While I understand this test was a bit inconvenient. Now, the API can return empty json and the test would pass.
Can we maybe create here an object with default config and compare that unmarshalled config is the same as default config?

Reply:
Is empty json a valid `kuma_cp.Config`? If so, what's wrong with the endpoint returning it?

We declare the default config already in one place. What's the point of a test where we just copy the default config and then check that _that_ declaration is the same as the first declaration?

Reply:
Actually you're right. It's unfortunate that we can even create a `kuma_cp.Config` that's empty but that's just the nature of go I guess.

Reply:
So the test API server doesn't match the default config because of e.g. different listening ports.

But I added a check that it's equal to the config the server is using.

I need to either add `omitempty` to EnvVars or add `EnvVars: map[string]string{}` to the default config. I think adding it to the default config makes sense, then we just eliminate the `nil` vs `map[string]string{}` problem.

WDYT?

Reply:
https://github.com/kumahq/kuma/pull/4756